aws/resource_aws_lightsail_load_balancer.go

```diff
@@ -0,0 +1,239 @@
+					validation.StringMatch(regexp.MustCompile(`^[a-zA-Z0-9_\-.]+[^._\-]$`), "must contain only alphanumeric characters, underscores, hyphens, and dots"),
+				),
+			},
+			"health_check_path": {
+				Type:     schema.TypeString,
+				Optional: true,
+				Default:  "/",
+			},
+			"instance_port": {
+				Type:         schema.TypeInt,
+				Required:     true,
+				ForceNew:     true,
+				ValidateFunc: validation.IntBetween(0, 65535),
+			},
+			"tags": tagsSchema(),
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"created_at": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"dns_name": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"protocol": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"public_ports": {
+				Type:     schema.TypeList,
+				Computed: true,
+				Elem:     &schema.Schema{Type: schema.TypeInt},
+			},
+		},
+	}
+}
+
+func resourceAwsLightsailLoadBalancerCreate(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).lightsailconn
+
+	req := lightsail.CreateLoadBalancerInput{
+		HealthCheckPath:  aws.String(d.Get("health_check_path").(string)),
+		InstancePort:     aws.Int64(int64(d.Get("instance_port").(int))),
+		LoadBalancerName: aws.String(d.Get("name").(string)),
+	}
+
+	if v := d.Get("tags").(map[string]interface{}); len(v) > 0 {
+		req.Tags = keyvaluetags.New(v).IgnoreAws().LightsailTags()
+	}
+
+	resp, err := conn.CreateLoadBalancer(&req)
+	if err != nil {
+		return err
+	}
+
+	if len(resp.Operations) == 0 {
+		return fmt.Errorf("No operations found for CreateInstance request")
+	}
+
+	op := resp.Operations[0]
+	d.SetId(d.Get("name").(string))
+
+	stateConf := &resource.StateChangeConf{
+		Pending:    []string{"Started"},
+		Target:     []string{"Completed", "Succeeded"},
+		Refresh:    resourceAwsLightsailLoadBalancerOperationRefreshFunc(op.Id, meta),
+		Timeout:    10 * time.Minute,
+		Delay:      5 * time.Second,
+		MinTimeout: 3 * time.Second,
+	}
+
+	_, err = stateConf.WaitForState()
+	if err != nil {
+		// We don't return an error here because the Create call succeeded
+		log.Printf("[ERR] Error waiting for load balancer (%s) to become ready: %s", d.Id(), err)
+	}
+
+	return resourceAwsLightsailLoadBalancerRead(d, meta)
+}
+
+func resourceAwsLightsailLoadBalancerRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*AWSClient).lightsailconn
+	ignoreTagsConfig := meta.(*AWSClient).IgnoreTagsConfig
+
+	resp, err := conn.GetLoadBalancer(&lightsail.GetLoadBalancerInput{
+		LoadBalancerName: aws.String(d.Id()),
+	})
+
+	if err != nil {
+		if awsErr, ok := err.(awserr.Error); ok {
+			if awsErr.Code() == "NotFoundException" {
+				log.Printf("[WARN] Lightsail load balancer (%s) not found, removing from state", d.Id())
+				d.SetId("")
+				return nil
+			}
+			return err
+		}
+		return err
+	}
+
```

Comment:
lb := resp.LoadBalancer

and reuse it for all others instead repeating it.

Reply:
Thanks for the feedback, this change will be implemented once the following pull request is merged by @akonrath https://github.com/akonrath/terraform-provider-aws/pull/96